include/atomics/receiver_cadmium.hpp

```diff
@@ -25,48 +35,82 @@
 
 #include "../data_structures/message.hpp"
 
+/** 
+* namespace is used to organize code into logical groups and to prevent name collisions.
+*/
 using namespace cadmium;
 using namespace std;
 
-//Port definition
+/**
+* Structure consists of class public output and input Messages
+* Port definition 
+*/
     struct receiver_defs {
 		struct out : public out_port<message_t> {
 		};
 		struct in : public in_port<message_t> {
 		};
 	};
    
+
     template<typename TIME>
+    
     class Receiver {
-        using defs = receiver_defs; // putting definitions in context
+
+        
+        using defs = receiver_defs;
+        /** 
+        * Putting reciever definitions in context. 
+        */  
         public:
-            //Parameters to be overwriten when instantiating the atomic model
+        
+        /** 
+        * Time constant Parameter
+        */
             TIME	PREPARATION_TIME;
-
-            // default constructor
+        
+            /** 
+            * receiver constructor.
+            * initializing acknowledgement to zero and sending to false values.
+            */
             Receiver() noexcept{
-              PREPARATION_TIME  = TIME("00:00:10");
+        
+                
+                
+            	PREPARATION_TIME  = TIME("00:00:10");
 				state.acknowledgement_num    = 0;
 				state.sending    = false;
             }
             
-            // state definition
+            /** 
+            * structure state definition and declaring acknowledgment number
+            */
             struct state_type {
 				int acknowledgement_num;
 				bool sending;
```

Comment:
lines 89 and 90 should include also brief description:
You need a brief description as follow:
int acknowledgement_num;   /**< description */ //!< Brief description after the member
bool sending;  /**< description */ //!< Brief description after the member